### PR TITLE
CCIP-4158 chain contract changeset

### PR DESCRIPTION
**Pull request by AnieeG**

### CODE REVIEW

File: deployment/ccip/changeset/deploy_chain.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"fmt"
+
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	ccipdeployment "github.com/smartcontractkit/chainlink/deployment/ccip"
+)
+
+var _ deployment.ChangeSet[ccipdeployment.DeployCCIPContractConfig] = InitialDeploy
```
Comment (connorwstein) [inline deployment/ccip/changeset/deploy_chain.go]: should be DeployChainContracts and have the DeployChainContractsConfig

**Review by connorwstein — APPROVED** [review]

File: deployment/ccip/changeset/deploy_chain.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	ccipdeployment "github.com/smartcontractkit/chainlink/deployment/ccip"
+)
+
+var _ deployment.ChangeSet[DeployChainContractsConfig] = DeployChainContracts
+
+func DeployChainContracts(env deployment.Environment, c DeployChainContractsConfig) (deployment.ChangesetOutput, error) {
```
Comment (makramkd) [inline deployment/ccip/changeset/deploy_chain.go:14]: Comments on these would be nice

File: deployment/ccip/changeset/deploy_chain.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	newAddresses := deployment.NewMemoryAddressBook()
+	err := ccipdeployment.DeployChainContractsForChains(env, newAddresses, c.HomeChainSelector, c.ChainSelectors, c.MCMSCfg)
+	if err != nil {
+		env.Logger.Errorw("Failed to deploy CCIP contracts", "err", err, "newAddresses", newAddresses)
+		return deployment.ChangesetOutput{AddressBook: newAddresses}, deployment.MaybeDataErr(err)
+	}
+	return deployment.ChangesetOutput{
+		Proposals:   []timelock.MCMSWithTimelockProposal{},
```
Comment (makramkd) [inline deployment/ccip/changeset/deploy_chain.go:22]: Are all of these changesets going to be converted to use MCMS? Or is MCMS for post-deployment processes only (transferring ownership, setting config, etc.)

File: deployment/ccip/changeset/deploy_chain_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	homeChainCfg := DeployHomeChainConfig{
+		HomeChainSel:     homeChainSel,
+		RMNStaticConfig:  ccdeploy.NewTestRMNStaticConfig(),
+		RMNDynamicConfig: ccdeploy.NewTestRMNDynamicConfig(),
+		NodeOperators:    ccdeploy.NewTestNodeOperator(e.Chains[homeChainSel].DeployerKey.From),
+		NodeP2PIDsPerNodeOpAdmin: map[string][][32]byte{
+			"NodeOperator": p2pIds,
+		},
+	}
+	output, err := DeployHomeChain(e, homeChainCfg)
```
Comment (makramkd) [inline deployment/ccip/changeset/deploy_chain_test.go:37]: Figuring out the correct order of incantations to get a functioning environment seems to be getting harder with all these new changesets, are we planning a more "hard to mess up" API in order to do things like this or nah?

File: deployment/ccip/changeset/jobspec.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import (
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	ccipdeployment "github.com/smartcontractkit/chainlink/deployment/ccip"
+)
+
+func Jobspec(env deployment.Environment, _ any) (deployment.ChangesetOutput, error) {
```
Comment (makramkd) [inline deployment/ccip/changeset/jobspec.go:11]: Comment + how this is expected to be used in conjunction with all the other stuff would be useful

Comment (makramkd) [reply deployment/ccip/changeset/jobspec.go:11]: Also naming is kinda not great on this one

File: deployment/ccip/deploy.go [hunk 6]
```diff
@@ -553,6 +530,62 @@ func DeployMCMSContracts(
...
+		return fmt.Errorf("capability registry not found")
+	}
+	cr, err := capReg.GetHashedCapabilityId(
+		&bind.CallOpts{}, CapabilityLabelledName, CapabilityVersion)
+	if err != nil {
+		e.Logger.Errorw("Failed to get hashed capability id", "err", err)
+		return err
+	}
+	if cr != CCIPCapabilityID {
+		return fmt.Errorf("capability registry does not support CCIP %s %s", hexutil.Encode(cr[:]), hexutil.Encode(CCIPCapabilityID[:]))
```
Comment (makramkd) [inline deployment/ccip/deploy.go:552]:
```suggestion
		return fmt.Errorf("unexpected mismatch between calculated ccip capability id (%s) and expected ccip capability id constant (%s)", hexutil.Encode(cr[:]), hexutil.Encode(CCIPCapabilityID[:]))
```

Comment (makramkd) [reply deployment/ccip/deploy.go:552]: This would ultimately be a bug in the constant

File: deployment/ccip/changeset/deploy_chain_test.go (hunk 4 above)
Comment (makramkd) [reply deployment/ccip/changeset/deploy_chain_test.go:37]: Or maybe for tests, a simple function that composes everything needed in order to get a functioning env

File: deployment/ccip/changeset/deploy_chain.go (hunk 3 above)
Comment (connorwstein) [reply deployment/ccip/changeset/deploy_chain.go:22]: MCMS for post deployment

File: deployment/ccip/changeset/deploy_chain_test.go (hunk 4 above)
Comment (connorwstein) [reply deployment/ccip/changeset/deploy_chain_test.go:37]: > a simple function that composes everything needed in order to get a functioning env

I added this in https://github.com/smartcontractkit/chainlink/pull/15288 FYI

**Review by mateusz-sekara — COMMENTED** [review]
LGTM, I don't have any comments but please fix @makramkd remarks first

**Review by makramkd — APPROVED** [review]

**Pull request merged**